src/highdicom/io.py

```diff
@@ -232,7 +241,7 @@ class ImageFileReader(object):
 
     """
 
-    def __init__(self, filename: Union[str, Path, DicomFileLike]):
+    def __init__(self, filename: Union[str, Path, DicomFileLike, DicomBytesIO]):
```

Comment:
Not needed because `DicomBytesIO` is a subtype of `DicomFileLike`.

```suggestion
    def __init__(self, filename: Union[str, Path, DicomFileLike]):
```